backend: ipmi: Make "mc reset" parameters configurable

okurz opened the pull request:
In some cases the connection to the IPMI interfaces is even less stable
than expected. With this change we are making all common parameters
configurable so that the test or instance admin can configure more and
be more graceful on connection problems.

Martchus commented on backend/ipmi.pm:
```diff
@@ -167,18 +167,20 @@ sub do_mc_reset {
+            my $count      = 0;
+            my $timeout    = $bmwqemu::vars{IPMI_MC_RESET_TIMEOUT}    // 60;
+            my $ping_count = $bmwqemu::vars{IPMI_MC_RESET_PING_COUNT} // 1;
+            my $ping_cmd   = "ping -c$ping_count " . $bmwqemu::vars{IPMI_HOSTNAME};
```
Not really related to your change, but it would make sense to quote the host name:
```suggestion
            my $ping_cmd   = "ping -c$ping_count '$bmwqemu::vars{IPMI_HOSTNAME}'";
```

okurz replied:
done

Martchus reviewed (commented):
I couldn't find any obvious mistakes. Nevertheless it would be great if the change was tested somehow.

okurz reviewed (commented):
ok, followed your suggestions and also did tests.

okurz commented:
Updated: There was one typo in the doc which failed t/04-check_vars_docu.t . Fixed and verified locally. We should not need to wait for travis CI if it takes very long.

Martchus commented on backend/ipmi.pm:
```diff
@@ -95,6 +95,7 @@ sub restart_host {
         last if $ret =~ m/is on/;
         $self->ipmitool('chassis power on');
     }
+    1;
```
Is this just a leftover?

okurz replied:
Actually I thought every good perl module should end with a success status but I could remove that without a failing test so I keep that change out of this PR. No problem.

Martchus replied:
But this is the middle of a function, not the end of the module :-)

okurz replied:
oh, wow. Well I have removed it because I misread your comment. That was stupid of me.

okurz replied:
https://github.com/os-autoinst/os-autoinst/pull/1587

By the way, because obviously we should not be blocked by hours-long waiting time for travis CI I continued with https://github.com/os-autoinst/os-autoinst/pull/1576

Martchus reviewed (commented):
Looks good, if the `1;` is just a leftover and removed, consider it approved.

Martchus commented on backend/ipmi.pm:
```diff
@@ -193,5 +196,3 @@ sub do_mc_reset {
 
     die "IPMI mc reset failure after $max_tries tries! Exit...";
 }
-
-1;
```
Likely this shouldn't have been removed.
